src/EditDialog.js

```diff
@@ -110,6 +111,26 @@ export default function EditDialog({ open, id, onClose }) {
 
   const dialogTitle = formatMessage(msgs.title);
 
+  useEffect(() => {
+    if (value) {
+      if (value.title) {
+        setTitle(value.title);
+      }
+
+      if (value.description) {
+        setDescription(value.description);
+      }
+
+      if (value.terms) {
+        setTerms(value.terms);
+      }
+
+      if (value.mapStyle) {
+        setMapStyle(value.mapStyle);
+      }
+    }
+  }, [value]);
+
```

Comment:
Im pretty sure this is redundant if you are using `onChange={e=>setState(e.target.value)}` on the `textfields`

Reply:
this is for the initialization of the text fields. The `value` refers to the map metadata that's retrieved from firebase

Reply:
Ahhh right, gotcha!